src/scala/com/github/johnynek/bazel_deps/GradleResolver.scala

```diff
@@ -42,203 +37,214 @@ class GradleResolver(
+          s"${dst.group.asString}:${dst.artifact.artifactId}"
+        )
+
+        missing(curEdge)
+      case None => false
+    }
+
+  def buildGraphFromDepMap(
     depMap: Map[String, GradleLockDependency]
-  ): Try[Graph[MavenCoordinate, Unit]] = {
-    assertConnectedDependencyMap(depMap).map(_ => cleanUpMap(depMap)).flatMap { depMap =>
-      def toCoord(k: String): Try[MavenCoordinate] =
-        depMap.get(k) match {
-          case Some(resolvedInfo) =>
-            val e = k.split(':')
-            val (org, nme) = (e(0), e(1))
+  ): Try[Graph[MavenCoordinate, Unit]] =
+    assertConnectedDependencyMap(depMap)
+      .map(_ => cleanUpMap(depMap))
+      .flatMap { depMap =>
+        type V[+A] = ValidatedNec[(String, String), A]
+        val cache = MMap[String, V[MavenCoordinate]]()
 
-            Success(
-              MavenCoordinate(
-                MavenGroup(org),
-                MavenArtifactId(
-                  nme,
-                  gradleTpe.getContentTypeOverride.getOrElse(k, "jar"),
-                  ""
-                ),
-                Version(resolvedInfo.locked.getOrElse(""))
-              )
-            )
-          case None =>
-            Failure(new Exception(s"Unable to lookup info about $k in dep map"))
-        }
+        def toCoord(k: String): V[MavenCoordinate] =
+          cache.getOrElseUpdate(k,
+            depMap.get(k) match {
+              case Some(resolvedInfo) =>
+                val e = k.split(':')
+                if (e.length >= 2) {
+                  val org = e(0)
+                  val nme = e(1)
 
-      val gradleDependencyGraphTry = depMap.toList.foldM(Graph.empty[MavenCoordinate, Unit]) {
-        case (g, (n, deps)) =>
-          for {
-            cnode <- toCoord(n)
-            transitive <- cats
-              .Traverse[List]
-              .sequence(deps.transitive.getOrElse(Nil).map(toCoord(_)))
-          } yield {
-            val g1 = g.addAllNodes(cnode :: transitive)
-            transitive.foldLeft(g1) { case (g, revDep) =>
-              val curEdge = (
-                s"${revDep.group.asString}:${revDep.artifact.artifactId}",
-                s"${cnode.group.asString}:${cnode.artifact.artifactId}"
+                  Validated.valid(MavenCoordinate(
+                    MavenGroup(org),
+                    MavenArtifactId(
+                      nme,
+                      gradleTpe.getContentTypeOverride.getOrElse(k, "jar"),
+                      ""
+                    ),
+                    Version(resolvedInfo.locked.getOrElse(""))
+                  ))
+                }
+                else {
+                  Validated.invalidNec((k, s"could not parse key $k into maven artifact"))
+                }
+              case None =>
+                Validated.invalidNec((k, s"Unable to lookup info about $k in dep map"))
+            })
+
+        depMap
+          .toList
+          .sortBy(_._1)
```

Comment:
Do you think it would be worth just using `SortedMap` consistently to ensure determinism, rather than having to periodically transform it? Maybe it's not worth it since you will still need `toList` before `traverse` but saving a `sortBy` might be worth it.

Reply:
I think that's a good future change.